core/src/data_sources/data_source.rs

```diff
@@ -805,6 +808,8 @@ impl DataSource {
                         let collection = self.qdrant_collection();
                         let chunk_size = self.config.max_chunk_size;
                         let qdrant_client = l_qdrant_client.clone();
+                        let mut token_count = chunks.len() * chunk_size;
+                        d.token_count = Some(token_count);
```

Comment:
Do we need that line??

Reply:
Yes, the expansion is interrupted early if there isn't enough space/chunks, so this handles that case

Reply:
ack :+1: